llvm/tools/llvm-profgen/ProfiledBinary.h

```diff
@@ -198,7 +198,14 @@ class ProfiledBinary {
   // Options used to configure the symbolizer
   symbolize::LLVMSymbolizer::Options SymbolizerOpts;
   // The runtime base address that the first executable segment is loaded at.
-  uint64_t BaseAddress = 0;
+  // The binary may be loaded at different addresses in different processes,
+  // so we use a map to store base address by PID.
+  // If the profile doesn't contain PID info we use the default PID value
+  // (DefaultPID defined in PerfReaderBase).
+  std::unordered_map<int32_t, uint64_t> BaseAddressByPID;
+  // The last PID we saw the binary loaded into. Used to warn the user if a
+  // binary is loaded in multiple processes without --multi-process-profile.
+  std::optional<int32_t> LastSeenPID = std::nullopt;
```

Comment:
Does this mean profiles from different processes needs to be ordered in a specific way?